src/ProtocolZoo/ProtocolZoo.jl

```diff
@@ -179,6 +179,10 @@ $TYPEDFIELDS
     rounds::Int = -1
     """maximum number of attempts to make per round (`-1` for infinite)"""
     attempts::Int = -1
+    """function or index of the slot to choose an available free slot in node A"""
+    chooseA::Union{Int,<:Function} = minimum
+    """function or index of the slot to choose an available free slot in node B"""
+    chooseB::Union{Int,<:Function} = minimum
     """whether the protocol should find the first available free slots in the nodes to be entangled or check for free slots randomly from the available slots"""
     randomize::Bool = false
```

Comment:
`randomize` and `chooseA` are now interdependent. I think it would be simpler to remove one from the struct and just use it in the constructor to potentially change what `chooseA` is.

Reply:
e.g. if the constructor is given `randomize=true`, then it overrides `chooseAslot` to be `rand`